File: app/src/test/java/com/mysticwind/linenotificationsupport/reply/impl/LocalizedMyPersonLabelProviderTest.java
package com.mysticwind.linenotificationsupport.reply.impl;

import static org.junit.Assert.*;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import android.content.Context;

import com.mysticwind.linenotificationsupport.ui.LocaleDao;

import org.junit.Test;

public class LocalizedMyPersonLabelProviderTest {

    @Test
    public void getMyPersonLabel_americanEnglishLabel() {
        LocalizedMyPersonLabelProvider classUnderTest = buildWithLocale("en-US");

        assertEquals("Me", classUnderTest.getMyPersonLabel().get());
    }

    @Test
    public void getMyPersonLabel_taiwanChineseLabel() {
        LocalizedMyPersonLabelProvider classUnderTest = buildWithLocale("zh-rTW");

        assertEquals("我", classUnderTest.getMyPersonLabel().get());
    }

    @Test
    public void getMyPersonLabel_defaultLabel() {
        LocalizedMyPersonLabelProvider classUnderTest = buildWithLocale("xxx");

        assertEquals("Me", classUnderTest.getMyPersonLabel().get());
    }

    private LocalizedMyPersonLabelProvider buildWithLocale(String locale) {
        LocaleDao localeDao = mock(LocaleDao.class);
        when(localeDao.getLocale()).thenReturn(locale);
        return new LocalizedMyPersonLabelProvider(localeDao, mock(Context.class));
    }

}

File: app/src/test/java/com/mysticwind/linenotificationsupport/reply/impl/LocalizedMyPersonLabelProviderTest.kt
package com.mysticwind.linenotificationsupport.reply.impl

import android.content.Context
import com.mysticwind.linenotificationsupport.ui.LocaleDao
import org.junit.Assert.assertEquals
import org.junit.Test
import org.mockito.kotlin.mock
import org.mockito.kotlin.whenever

class LocalizedMyPersonLabelProviderTest {

    @Test
    fun getMyPersonLabel_americanEnglishLabel() {
        val classUnderTest = buildWithLocale("en-US")

        assertEquals("Me", classUnderTest.getMyPersonLabel().get())
    }

    @Test
    fun getMyPersonLabel_taiwanChineseLabel() {
        val classUnderTest = buildWithLocale("zh-rTW")

        assertEquals("我", classUnderTest.getMyPersonLabel().get())
    }

    @Test
    fun getMyPersonLabel_defaultLabel() {
        val classUnderTest = buildWithLocale("xxx")

        assertEquals("Me", classUnderTest.getMyPersonLabel().get())
    }

    private fun buildWithLocale(locale: String): LocalizedMyPersonLabelProvider {
        val localeDao = mock<LocaleDao>()
        whenever(localeDao.getLocale()).thenReturn(locale)
        return LocalizedMyPersonLabelProvider(localeDao, mock<Context>())
    }
}
